subprojects/ide/src/main/java/org/gradle/plugins/ide/internal/resolver/IdeDependencySet.java

```diff
@@ -128,6 +128,8 @@ private ArtifactCollection getResolvedArtifacts(Configuration configuration, fin
                 public void execute(ArtifactView.ViewConfiguration viewConfiguration) {
                     viewConfiguration.lenient(true);
                     viewConfiguration.componentFilter(getComponentFilter(visitor));
+                    // TODO: Doing the restriction this way breaks a lot of test. The resulting start containing RUNTIME + TEST instead of COMPILE scopes for Idea for example.
+                    //viewConfiguration.attributes(attrs -> attrs.attribute(Usage.USAGE_ATTRIBUTE, NamedObjectInstantiator.INSTANCE.named(Usage.class, Usage.JAVA_RUNTIME_JARS)));
```

Comment:
Is there a reason why this breaks so many tests? Also is this change (COMPILE -> RUNTIME + TEST) limited to the tests or is this a breaking change for idea users? Since I'm not sure of the above I went ahead and did the filtering  in the `EclipseDependenciesCreator`

Reply:
You can't limit to runtime jars here, since compile dependencies are not jars, but classes folders.